ddl/resource_group.go

```diff
@@ -52,10 +61,17 @@ func onCreateResourceGroup(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64,
 		if err != nil {
 			return ver, errors.Trace(err)
 		}
-		err = infosync.AddResourceGroup(context.TODO(), protoGroup)
+
+		ctx, cancel := context.WithTimeout(d.ctx, defaultInfosyncTimeout)
+		defer cancel()
+		err = infosync.AddResourceGroup(ctx, protoGroup)
 		if err != nil {
 			logutil.BgLogger().Warn("create resource group failed", zap.Error(err))
-			return ver, errors.Trace(err)
+			// TiDB will add the group to the resource manager when it bootstraps.
+			// add judge here to compatible with keyspace mode TiDB to skip the exist error with default group.
+			if !strings.Contains(err.Error(), alreadyExists) || groupInfo.Name.L != defaultGroupName {
```

Comment:
If the group exists, then this DDL event should be a noop?
```suggestion
			if !strings.Contains(err.Error(), alreadyExists) {
```

Reply:
expect the `default` group, we need to report the already exist error.

Reply:
We should check the `IF NOT EXISTS` flag if encountering the `alreadyExists` error.

Reply:
what do you mean here? I think it's no problem because the `IF NOT EXISTS` only check in kv schema side, the problem is check in pd side.